README.md

```diff
@@ -160,6 +179,57 @@ describe 'myclass' do
 end
 ```
 
+When using roles and profiles to manage a heterogeneous IT estate, you can test a profile that supports several OSes with many `let(:facts)` as long as each is in its own context:
+```ruby
+require 'spec_helper'
+
+describe 'profiles::packagerepos' do
+  context 'Raspbian tests' do # We manage hundreds of desk-mounted Pis
+    raspbian = {
+      hardwaremodels: ['armv7l'],
+      supported_os: [
+        {
+          'operatingsystem'        => 'Debian',
+          'operatingsystemrelease' => ['10', '9', '8'],
+        },
+      ],
+    }
+    on_supported_os(raspbian).each do |os, os_facts|
+      let(:facts) do
+        os_facts
+      end
+
+      context "#{os} with defaults" do
+        it { is_expected.to compile }
+        # more tests ...
+      end
+    end
+  end
+  context 'Ubuntu tests' do # And also a fleet of Ubuntu desktops
+    ubuntu = {
+      supported_os: [
+        {
+          'operatingsystem'        => 'Ubuntu',
+          'operatingsystemrelease' => ['18.04', '16.04'],
+        },
+      ],
+    }
+
+    on_supported_os(ubuntu).each do |os, os_facts|
+      let(:facts) do
+        os_facts
+      end
+
+      context "#{os} with defaults" do
+        it { is_expected.to compile }
+        # more tests ...
+      end
+    end
+    # Rinse and repeat with CentOS...
```

Comment:
This might be misunderstood as rspec-puppet-facts iterating over CentOS. Suggest removing this as the example above already makes it clear how to add other systems.

Reply:
Nuked.